mirgecom/artificial_viscosity.py

```diff
@@ -228,56 +237,65 @@ def av_laplacian_operator(discr, boundaries, fluid_state, alpha, gas_model=None,
     interp_to_surf_quad = partial(tracepair_with_discr_tag, discr, quadrature_tag)
 
     def interp_to_vol_quad(u):
-        return op.project(discr, "vol", dd_vol, u)
+        return op.project(discr, dd_base, dd_vol, u)
 
     if operator_states_quad is None:
         from mirgecom.gas_model import make_operator_fluid_states
         operator_states_quad = make_operator_fluid_states(
-            discr, fluid_state, gas_model, boundaries, quadrature_tag)
+            discr, fluid_state, gas_model, boundaries, quadrature_tag,
+            volume_dd=dd_base)
 
     vol_state_quad, inter_elem_bnd_states_quad, domain_bnd_states_quad = \
         operator_states_quad
 
     # Get smoothness indicator based on mass component
     if indicator is None:
         indicator = smoothness_indicator(discr, fluid_state.mass_density,
-                                         kappa=kappa, s0=s0)
+                                         kappa=kappa, s0=s0, volume_dd=dd_base)
 
     if grad_cv is None:
         from mirgecom.navierstokes import grad_cv_operator
         grad_cv = grad_cv_operator(discr, gas_model, boundaries, fluid_state,
                                    time=time, quadrature_tag=quadrature_tag,
+                                   volume_dd=dd_base,
                                    operator_states_quad=operator_states_quad)
 
     # Compute R = alpha*grad(Q)
     r = -alpha * indicator * grad_cv
 
     def central_flux_div(utpair):
-        dd = utpair.dd
-        normal = actx.thaw(discr.normal(dd))
-        return op.project(discr, dd, dd.with_dtag("all_faces"),
+        dd_trace = utpair.dd
+        dd_allfaces = dd_trace.with_domain_tag(
+            replace(dd_trace.domain_tag, tag=FACE_RESTR_ALL))
```

Comment:
This is clunky. :shrug:

Reply:
Fixed in inducer/grudge#280 with `with_boundary_tag`.